File: src/model/endpoint.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

pub type EnvList = HashMap<String, String>;
#[derive(Serialize, Deserialize)]
pub struct Endpoint {
    pub path: String,
    pub desc: String,
    pub method: Option<String>,
}

type EndpointName = String;

type EndpointList = HashMap<EndpointName, Endpoint>;

// $defsを利用しているプロジェクトがあるので仕方なくオプションにしている
#[derive(Serialize, Deserialize)]
pub struct EndpointAssets {
    pub env: Option<EnvList>,
    pub api: Option<EndpointList>,
}

type EndpointVersion = String;

pub type EndpointSetting = HashMap<EndpointVersion, EndpointAssets>;

File: src/model/endpoint.ts
export type EnvList = Record<string, string>;

export type AuthSchema =
  | {
      type: "Bearer" | "Basic";
      header: "Authorization";
    }
  | {
      type: "ApiKey";
      header: "X-Access-Token";
    };

export interface Endpoint {
  path: string;
  desc: string;
  method?: string;
  authSchema?: AuthSchema;
  request?: null | Record<string, unknown>;
  response?: null | Record<string, unknown>;
}

type EndpointList = Record<string, Endpoint>;

export interface EndpointAssets {
  env?: EnvList;
  api?: EndpointList;
}

export type EndpointSetting = Record<string, EndpointAssets>;
